onnxruntime/core/optimizer/attention_fusion_helper.h

```diff
@@ -777,10 +839,26 @@ bool CheckNodesInPathV(const Graph& graph, const Node& reshape, const Node& tran
 
   // Check reshape for attention output has shape input (0, 0, -1) or (0, 0, N*H)
   std::vector<int64_t> reshape_shape;
-  if (!optimizer_utils::AppendTensorFromInitializer(graph, *(reshape.InputDefs()[1]), reshape_shape) ||
-      reshape_shape.size() != 3 ||
+  if (!optimizer_utils::AppendTensorFromInitializer(graph, *(reshape.InputDefs()[1]), reshape_shape)) {
```

Comment:
These code is not needed. Assume that reshape fusion has done this, the shape of reshape is now an initializer.